### PR TITLE
fix: specify branch name as env image tag

### PR DESCRIPTION
**Pull request by STRRL**
Signed-off-by: STRRL <str_ruiling@outlook.com>

### What problem does this PR solve?

GitHub Action failed to build images for release: https://github.com/chaos-mesh/chaos-mesh/actions/runs/1633015443

### What's changed and how it works?

Use the dev-env and build-env with the branch `release-2.1`

### Related changes

- [ ] Need to update `chaos-mesh/website`
- [ ] Need to update `Dashboard UI`
- Need to **cheery-pick to release branches**
  - [x] release-2.1
  - [ ] release-2.0

### Checklist

Tests

- [ ] Unit test
- [ ] E2E test
- [x] No code
- [ ] Manual test (add steps below)

Side effects

- [ ] Breaking backward compatibility

### Release note 

```text
Please add a release note.

You can safely ignore this section if you don't think this PR needs a release note.
```

### DCO

If you find the DCO check fails, please run commands like below (Depends on the actual situations. For example, if the failed commit isn't the most recent) to fix it:

```shell
git commit --amend --signoff
git push --force
```

### CODE REVIEW

**Comment by STRRL** [comment]
PTAL

/cc @chaos-mesh/committers @chaos-mesh/maintainers

**Review by YangKeao — APPROVED** [review]
If the `build-env` and `dev-env` is modified, and this action is run for an older release, it will fail.

But it's not a big problem, I prefer to merge this PR before considering that situation 😃 .

File: .github/workflows/upload_image.yml [hunk 1]
```diff
@@ -16,17 +16,31 @@ jobs:
...
+      - uses: actions/checkout@v2
+        with:
+          # it is required for fetching branches which contains certain tag
+          fetch-depth: 0
 
-      - name: Extract Image Tag
+      - name: Extract Image Tags
         shell: bash
         run: |
           IMAGE_TAG=${GITHUB_REF##*/}
```
Comment (g1eny0ung) [inline .github/workflows/upload_image.yml:27]: This can be replaced by `${{ github.ref_name }}`.

**Comment by STRRL** [comment]
> If the build-env and dev-env is modified, and this action is run for an older release, it will fail.

I think we would only build once for a certain release, so here is no situation for using modified build-env and dev-env for building older releases.

Or should we maintain `build-env` or `dev-env` for the branch, or the tagged commit? Which one would you choose? 🤔

File: .github/workflows/upload_image.yml (hunk 1 above)
Comment (STRRL) [reply .github/workflows/upload_image.yml]: Updated! PTAL

**Comment by STRRL** [comment]
> Or should we maintain `build-env` or `dev-env` for the tagged commit instead of branch? Which one would you choose?

Or none of then, set `IMAGE_DEV_ENV_BUILD=1` and `IMAGE_BUILD_ENV_BUILD=1` in `uplad_image.yml`?

**Comment by g1eny0ung** [comment]
> I think we would only build once for a certain release, so here is no situation for using modified build-env and dev-env for building older releases.

@STRRL I prefer this.

> > Or should we maintain `build-env` or `dev-env` for the tagged commit instead of branch? Which one would you choose?
> 
> Or none of then, set `IMAGE_DEV_ENV_BUILD=1` and `IMAGE_BUILD_ENV_BUILD=1` in `uplad_image.yml`?

I thought of this too, do we need to build these two mirrors at release?

**Comment by g1eny0ung** [comment]
> I think we would only build once for a certain release, so here is no situation for using modified build-env and dev-env for building older releases.

@STRRL I think this is more intuitive.

> > Or should we maintain `build-env` or `dev-env` for the tagged commit instead of branch? Which one would you choose?
> 
> Or none of then, set `IMAGE_DEV_ENV_BUILD=1` and `IMAGE_BUILD_ENV_BUILD=1` in `uplad_image.yml`?

I thought of this too, do we need to build these two images at release?

**Comment by STRRL** [comment]
> I thought of this too, do we need to build these two images at release?

Currently, we maintain `dev-env` and `build-env` for each branch: `master` and `release-2.1`. When releasing `2.1.x`,  it should use the image `dev-env:release-2.1` and `build-env:release-2.1`, it does not build these two images at release.

**Comment by YangKeao** [comment]
> > If the build-env and dev-env is modified, and this action is run for an older release, it will fail.
> 
> I think we would only build once for a certain release, so here is no situation for using modified build-env and dev-env for building older releases.

Yes. As we will never "release again" a tag, it's fine to assume that the latest `build-env:v2.1` is the one built for release 2.1.2 (though this assumption cannot be true for a long time.)

I prefer to build a release image with `IMAGE_DEV_ENV_BUILD=1` and `IMAGE_BUILD_ENV_BUILD=1`, which sounds cleaner (though, there is no actual benefit 😃 )

**Comment by STRRL** [comment]
> I prefer to build a release image with `IMAGE_DEV_ENV_BUILD=1` and `IMAGE_BUILD_ENV_BUILD=1`, which sounds cleaner (though, there is no actual benefit  )

+1

I am going to close this PR and create another one for applying these changes.

**Pull request closed without merging**